services/order.py

```diff
@@ -0,0 +1,46 @@
+from django.contrib.auth import get_user_model
+from django.db import transaction
+from django.db.models import QuerySet
+
+import init_django_orm  # noqa: F401
+from db.models import Ticket, Order, User
+
+
+@transaction.atomic
+def create_order(tickets: list[dict], username: str, date: str = None) -> None:
```

Comment:
Consider using `Optional[str]` for the `date` parameter to explicitly indicate that it can be `None`. Additionally, using a `datetime` object instead of a string for the date would ensure proper handling and validation of date formats.

Reply:
We expect a string, not a datetime object.